core/src/test/scala/akka/kafka/scaladsl/ReconnectSpec.scala

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright (C) 2014 - 2016 Softwaremill <http://softwaremill.com>
+ * Copyright (C) 2016 - 2018 Lightbend Inc. <http://www.lightbend.com>
+ */
+
+package akka.kafka.scaladsl
+
+import akka.kafka._
+import akka.kafka.test.Utils._
+import akka.stream.{KillSwitches, UniqueKillSwitch}
+import akka.stream.scaladsl.{Keep, Sink, Tcp}
+import net.manub.embeddedkafka.EmbeddedKafkaConfig
+
+import scala.concurrent.{Await, Future}
+import scala.concurrent.duration._
+
+class ReconnectSpec extends SpecBase(kafkaPort = KafkaPorts.ReconnectSpec) {
+
+  val proxyPort = KafkaPorts.ReconnectSpecProxy
+
+  def createKafkaConfig: EmbeddedKafkaConfig =
+    EmbeddedKafkaConfig(
+      kafkaPort, zooKeeperPort,
+      Map(
+        "offsets.topic.replication.factor" -> "1"
+      ))
+
+  "A killed Kafka server" must {
+
+    "work for a Producer" in assertAllStagesStopped {
+      val topic1 = createTopic(1)
+      val group1 = createGroup(1)
+
+      givenInitializedTopic(topic1)
+
+      val producerSettings = producerDefaults.withBootstrapServers(s"localhost:$proxyPort")
+
+      // start a TCP proxy forwarding to Kafka and start producing messages
+      val (proxyBinding, proxyKillSwtich) = createProxy()
+      Await.ready(proxyBinding, remainingOrDefault)
+
+      val messagesProduced = 100
+      produce(topic1, 1 to messagesProduced, producerSettings)
```

Comment:
It looks like all of the messages will be sent already before the producer proxy connection is killed. Or am I missing something?

Reply:
Yes, you are right. The approach was too naïve.